### PR TITLE
Tag the Consumer before publishing the Pact to the Broker

### PR DESCRIPTION
**Pull request by enesanbar**
Closes: #131

### CODE REVIEW

File: src/PhpPact/Consumer/Listener/PactTestListener.php [hunk 1]
```diff
@@ -118,8 +118,8 @@ public function endTestSuite(TestSuite $suite): void
                 $client = new GuzzleClient($clientConfig);
 
                 $brokerHttpService = new BrokerHttpClient($client, new Uri($pactBrokerUri));
-                $brokerHttpService->publishJson($json, $consumerVersion);
                 $brokerHttpService->tag($this->mockServerConfig->getConsumer(), $consumerVersion, $tag);
+                $brokerHttpService->publishJson($json, $consumerVersion);
```
Comment (cfmack) [inline src/PhpPact/Consumer/Listener/PactTestListener.php]: What if this is a new consumer?   Does tag create the Consumer?   Does the broker create the consumer in the tag function before there is an initial contract?

Comment (enesanbar) [reply src/PhpPact/Consumer/Listener/PactTestListener.php]: Yes, broker takes care of that. "tag" function just makes a PUT request to this path:
"/pacticipants/{$consumer}/versions/{$version}/tags/{$tag}", which in turn creates the consumer if it doesn't exist.

```sh
postgres=# select * from pacticipants;
 id | name | repository_url | created_at | updated_at
----+------+----------------+------------+------------
(0 rows)
```

```sh
$ curl -X PUT -H "Content-Type: application/json" http://pact-broker/pacticipants/non-existing-consumer/versions/1.0.0/tags/master/ -i
HTTP/1.1 201 Created
Content-Type: application/hal+json;charset=utf-8
Content-Length: 510
Connection: keep-alive
Status: 201 Created
Location: http://pact-broker/pacticipants/non-existing-consumer/versions/1.0.0/tags/master
Date: Thu, 20 Jun 2019 05:47:35 GMT
X-Content-Type-Options: nosniff
Server: Webmachine-Ruby/1.5.0 Rack/1.3
X-Pact-Broker-Version: 2.31.0
X-Powered-By: Phusion Passenger 6.0.2

{"name":"master","updatedAt":"2019-06-20T05:47:35+00:00","createdAt":"2019-06-20T05:47:35+00:00","_links":{"self":{"title":"Tag","name":"master","href":"http://pact-broker/pacticipants/non-existing-consumer/versions/1.0.0/tags/master"},"version":{"title":"Version","name":"1.0.0","href":"http://pact-broker/pacticipants/non-existing-consumer/versions/1.0.0"},"pacticipant":{"title":"Pacticipant","name":"non-existing-consumer","href":"http://pact-broker/pacticipants/non-existing-consumer"}}}
```

```sh
postgres=# select * from pacticipants;
 id |         name          | repository_url |         created_at         |         updated_at
----+-----------------------+----------------+----------------------------+----------------------------
  3 | non-existing-consumer |                | 2019-06-20 05:47:35.663317 | 2019-06-20 05:47:35.663334
(1 row)
```

**Comment by bethesque** [comment]
The tag was always designed to come first (autocreating the consumer), but for an unintended reason, it used to handle it ok if the tag came second. I made a change recently that means that the tag has to come first, and so it's brought to light all the libraries that had the ordering the wrong way around. I didn't realise there were so many places not using the "official" pact-broker-client.

If I was designing the API 6 years ago knowing what I know now, it wouldn't have this ordering dependency! You live and learn.

**Comment by cfmack** [comment]
Thanks y'all

**Pull request merged**